pkg/descheduler/node/node.go

```diff
@@ -323,3 +330,18 @@ func PodMatchNodeSelector(pod *v1.Pod, node *v1.Node) bool {
 	}
 	return matches
 }
+
+// podMatchesInterPodAntiAffinity checks if the pod matches the anti-affinity rule
+// of another pod that is already on the given node.
+// If a match is found, it returns true.
+func podMatchesInterPodAntiAffinity(nodeIndexer podutil.GetPodsAssignedToNodeFunc, pod *v1.Pod, node *v1.Node) (bool, error) {
+	podsOnNode, err := podutil.ListPodsOnANode(node.Name, nodeIndexer, nil)
```

Comment:
Can we get out early. In a cluster with 20k pods and only a handful of pods with PodAntiAffinity this can get expensive, right?


```suggestion
        if pod.Spec.Affinity == nil || pod.Spec.Affinity. PodAntiAffinity == nil {
          return false, nil
        }
	podsOnNode, err := podutil.ListPodsOnANode(node.Name, nodeIndexer, nil)
```

Reply:
Absolutely - just added this check in the last commit! Thanks @a7i 👍